lib/rules/no-duplicate-case.js

```diff
@@ -31,18 +37,31 @@ module.exports = {
     create(context) {
         const sourceCode = context.getSourceCode();
 
+        /**
+         * Determines whether the two given nodes are considered to be equal.
+         * @param {ASTNode} a First node.
+         * @param {ASTNode} b Second node.
+         * @returns {boolean} `true` if the nodes are considered to be equal.
+         */
+        function equal(a, b) {
+            if (a.type !== b.type) {
+                return false;
+            }
```

Comment:
Just a thought, 

I think this check can be moved to `astUtils.equalTokens` itself as a point of an early return.

@mdjermanovic thoughts ?

Reply:
That would be certainly good for optimization, but we should then probably also change the name and the description of this function (or maybe add a new function that would check the types first and then call this one). In theory, `equalTokens` check might be used to compare two different node types (e.g. a `FunctionDeclaration` and a `FunctionExpression` can have equal tokens). Maybe we could discuss this separated from this PR?

Reply:
Agreed that we should decouple that discussion from this PR.

Reply:
cool